### PR TITLE
Stage templates manifests

### PR DESCRIPTION
**Pull request by cd1989**
**What this PR does / why we need it**:

Add your description

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #
https://github.com/caicloud/cyclone/issues/706

**Special notes for your reviewer**:

/cc @supereagle @zhujian7 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: manifests/templates/cicd/build.yaml [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        Build Java source code with Maven
+  spec:
+    pod:
+      inputs:
+        arguments:
+        - name: image
+          value: maven:3.5.3-jdk-8-alpine
+        - name: cmd
+          value: |
+            cd /workspace/<repo-name> &&
```
Comment (supereagle) [inline manifests/templates/cicd/build.yaml]: Directly git clone into `/workspace`? Then users do not need to car about `repo-name`.

File: manifests/templates/cicd/code-scan.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      cyclone.io/scene: cicd
+      cyclone.io/template-kind: code-scan
+    annotations:
+      cyclone.io/description: >
+        Source code statistically scan with SonarQube
+  spec:
+    pod:
+      inputs:
+        arguments:
+        - name: sonarqube
```
Comment (supereagle) [inline manifests/templates/cicd/code-scan.yaml]: Where will use this argument?

File: manifests/templates/cicd/code-scan.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      outputs:
+        resources:
+        - name: image
+      spec:
+        containers:
+        - image: "{{ image }}"
+          command:
+          - /bin/sh
+          - -c
+          - "{{ cmd }}"
```
Comment (supereagle) [inline manifests/templates/cicd/code-scan.yaml]: Add newline

File: pkg/server/biz/templates/stages.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
+package templates
```
Comment (supereagle) [inline pkg/server/biz/templates/stages.go]: What does `biz` mean?

File: manifests/templates/cicd/build.yaml [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+  spec:
+    pod:
+      inputs:
+        arguments:
+        - name: image
+          value: golang:1.10-alpine3.8
+        - name: cmd
+          value: |
+            # Replace <main-entrypoint-path> with the main package path, for example 'cyclone/cmd/workflow/controller'.
+            go build -o /go/bin/app <main-entrypoint-path>
```
Comment (zhujian7) [inline manifests/templates/cicd/build.yaml]: Can you explain why we need `-o /go/bin/app`, I think the output path should be specified by users.

File: manifests/templates/cicd/build.yaml (hunk 1 above)
Comment (zhujian7) [reply manifests/templates/cicd/build.yaml]: Agree, Java doesn't care about `repo-name`.

File: manifests/templates/cicd/build.yaml (hunk 5 above)
Comment (zhujian7) [reply manifests/templates/cicd/build.yaml]: My bad, I found you give the argument `output-artifact-path` a default value `/go/bin/app`. That is nice.

File: manifests/templates/cicd/code-scan.yaml (hunk 2 above)
Comment (zhujian7) [reply manifests/templates/cicd/code-scan.yaml]: This argument is used for Users to choose a sonarqube instance from Integration.

File: manifests/templates/cicd/image-build.yaml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        - name: cmd
+          value: |
+            cd /workspace/<repo-name> &&
+            docker build -f {{ dockerfile-path }} -t {{ image-name }} {{ build-context }}
+        resources:
+        - name: code
+          path: /workspace
+      outputs:
+        resources:
+        - name: image
```
Comment (zhujian7) [inline manifests/templates/cicd/image-build.yaml]: Can we give the `image` a default value `{{ image-name }}`?

File: manifests/templates/cicd/build.yaml [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+      outputs:
+        artifacts:
+        - name: bin
+          path: "{{ output-artifact-path }}"
+      spec:
+        containers:
+        - image: "{{ image }}"
+          command:
+          - /bin/sh
+          - -c
```
[the remark this answers is not in the document]
Comment (cd1989) [reply manifests/templates/cicd/build.yaml]: ok

File: manifests/templates/cicd/image-build.yaml (hunk 6 above)
Comment (cd1989) [reply manifests/templates/cicd/image-build.yaml]: Image name is specified in the image resource definition, here is ref to the image resource

File: pkg/server/biz/templates/stages.go (hunk 4 above)
Comment (cd1989) [reply pkg/server/biz/templates/stages.go]: business

File: manifests/templates/cicd/build.yaml (hunk 1 above)
Comment (cd1989) [reply manifests/templates/cicd/build.yaml]: Resolved. https://github.com/caicloud/cyclone/pull/739/files

**Comment by cd1989** [comment]
> Is it necessary to hold these templates configuration in a configmap?
I'm afraid this CM would growth too large when more templates added.

**Comment by supereagle** [comment]
> Is it necessary to hold these templates configuration in a configmap?

I think they can be created while install Cyclone, we can put them in [manifests](https://github.com/caicloud/cyclone/tree/master/manifests).

**Pull request merged**